packages/bitcore-wallet-service/src/lib/chain/eth/index.ts

```diff
@@ -112,48 +96,84 @@ export class EthChain implements IChain {
 
   getChangeAddress() {}
 
-  checkDust(output, opts) {
-    if (opts.outputs.length != 1) {
-      return Errors.MORE_THAT_ONE_OUTPUT;
-    }
-  }
+  checkDust(output, opts) {}
 
   getFee(server, wallet, opts) {
     return new Promise(resolve => {
-      server._getFeePerKb(wallet, opts, (err, inFeePerKb) => {
+      server._getFeePerKb(wallet, opts, async (err, inFeePerKb) => {
         let feePerKb = inFeePerKb;
         let gasPrice = inFeePerKb;
-        const { from, data, outputs } = opts;
+        const { from } = opts;
         const { coin, network } = wallet;
-        server.estimateGas(
-          {
-            coin,
-            network,
-            from,
-            to: outputs[0].toAddress,
-            value: outputs[0].amount,
-            data,
-            gasPrice
-          },
-          (err, inGasLimit) => {
-            if (_.isNumber(opts.fee)) {
-              // This is used for sendmax
-              gasPrice = feePerKb = Number(
-                (
-                  opts.fee / (inGasLimit || Defaults.DEFAULT_GAS_LIMIT)
-                ).toFixed()
-              );
-            }
-
-            const gasLimit = inGasLimit || Defaults.DEFAULT_GAS_LIMIT;
-            opts.fee = feePerKb * gasLimit;
-            return resolve({feePerKb, gasPrice, gasLimit});
+        let inGasLimit;
+        for (let output of opts.outputs) {
+          try {
+            inGasLimit = await server.estimateGas({
+              coin,
+              network,
+              from,
+              to: output.toAddress,
+              value: output.amount,
+              data: output.data,
+              gasPrice
+            });
+            output.gasLimit = inGasLimit || Defaults.DEFAULT_GAS_LIMIT;
+          } catch (err) {
+            output.gasLimit = Defaults.DEFAULT_GAS_LIMIT;
           }
-        );
+        }
+        if (_.isNumber(opts.fee)) {
+          // This is used for sendmax
+          gasPrice = feePerKb = Number(
+            (opts.fee / (inGasLimit || Defaults.DEFAULT_GAS_LIMIT)).toFixed()
+          );
+        }
+
+        const gasLimit = inGasLimit || Defaults.DEFAULT_GAS_LIMIT;
+        opts.fee = feePerKb * gasLimit;
+        return resolve({ feePerKb, gasPrice });
       });
     });
   }
 
+  buildTx(txp) {
+    const isERC20 = txp.tokenAddress && !txp.payProUrl;
+    const chain = isERC20 ? 'ERC20' : 'ETH';
```

Comment:
chain = `ERC20' is very difficult to understand. Maybe Transaction.create param should be renamed to `coin` or `coinType`

Reply:
Agree, chain = ERC20 is not ideal